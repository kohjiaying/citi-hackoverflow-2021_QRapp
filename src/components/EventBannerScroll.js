import React, { Component } from "react"
import {
  ScrollView,
  Animated,
  SafeAreaView,
  ImageBackground,
  Dimensions,
} from "react-native"

const OFFSET = 40
const ITEM_WIDTH = Dimensions.get("window").width - (OFFSET * 2)
const ITEM_HEIGHT = 200

const cards = [
  { title: "Movie 1", posterUrl: require("../assets/images/tenent.jpg") },
  { title: "Movie 2", posterUrl: require("../assets/images/tenent.jpg") },
  { title: "Movie 3", posterUrl: require("../assets/images/tenent.jpg") },
  { title: "Movie 4", posterUrl: require("../assets/images/tenent.jpg") },
]

export default function EventBannerScroll() {
  const scrollX = React.useRef(new Animated.Value(0)).current

  return (
      <ScrollView
        horizontal={true}
        decelerationRate={"normal"}
        snapToInterval={ITEM_WIDTH}
        style={{ marginTop: 40, paddingHorizontal: 0 }}
        showsHorizontalScrollIndicator={false}
        bounces={false}
        disableIntervalMomentum
        onScroll={Animated.event(
          [{ nativeEvent: { contentOffset: { x: scrollX } } }],
          { useNativeDriver: false }
        )}
        scrollEventThrottle={12}
      >
        {cards.map((item, idx) => {
          const inputRange = [
            (idx - 1) * ITEM_WIDTH,
            idx * ITEM_WIDTH,
            (idx + 1) * ITEM_WIDTH,
          ]

          const translate = scrollX.interpolate({
            inputRange,
            outputRange: [0.85, 1, 0.85],
          })

          const opacity = scrollX.interpolate({
            inputRange,
            outputRange: [0.5, 1, 0.5],
          })

          return (
            <Animated.View
			  key = {item.title}
              style={{
                width: ITEM_WIDTH,
                height: ITEM_HEIGHT,
                marginLeft: idx === 0 ? OFFSET : undefined,
                marginRight: idx === cards.length - 1 ? OFFSET : undefined,
                opacity: opacity,
                transform: [{ scale: translate }],
              }}
            >
              <ImageBackground
                source={item.posterUrl}
                style={{
                  flex: 1,
                  resizeMode: "cover",
                  justifyContent: "center",
                }}
                imageStyle={{ borderRadius: 6}}
              />
            </Animated.View>
          )
        })}
      </ScrollView>
  )
}