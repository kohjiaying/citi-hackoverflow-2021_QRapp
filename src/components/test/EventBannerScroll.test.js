import { JestEnvironment } from '@jest/environment';
import React from 'react';
import renderer from 'react-test-renderer';
import EventBannerScroll from '../EventBannerScroll';

describe('<EventBannerScroll />', () => {
    it('renders correctly', () => {
      const tree = renderer.create(<EventBannerScroll />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });