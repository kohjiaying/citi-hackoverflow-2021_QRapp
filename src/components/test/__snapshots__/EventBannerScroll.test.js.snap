// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventBannerScroll /> renders correctly 1`] = `
<RCTScrollView
  bounces={false}
  decelerationRate="normal"
  disableIntervalMomentum={true}
  horizontal={true}
  onScroll={[Function]}
  scrollEventThrottle={12}
  showsHorizontalScrollIndicator={false}
  snapToInterval={670}
  style={
    Object {
      "marginTop": 20,
      "paddingHorizontal": 0,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "height": 200,
          "marginLeft": 40,
          "marginRight": undefined,
          "opacity": 1,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
          "width": 670,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "resizeMode": "cover",
          }
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
              Object {
                "borderRadius": 6,
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "height": 200,
          "marginLeft": undefined,
          "marginRight": undefined,
          "opacity": 0.5,
          "transform": Array [
            Object {
              "scale": 0.85,
            },
          ],
          "width": 670,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "resizeMode": "cover",
          }
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
              Object {
                "borderRadius": 6,
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "height": 200,
          "marginLeft": undefined,
          "marginRight": undefined,
          "opacity": 0,
          "transform": Array [
            Object {
              "scale": 0.7,
            },
          ],
          "width": 670,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "resizeMode": "cover",
          }
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
              Object {
                "borderRadius": 6,
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "height": 200,
          "marginLeft": undefined,
          "marginRight": 40,
          "opacity": -0.5,
          "transform": Array [
            Object {
              "scale": 0.5499999999999999,
            },
          ],
          "width": 670,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "resizeMode": "cover",
          }
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
              Object {
                "borderRadius": 6,
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
